File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int countNodes(TreeNode* root) {
        if ( !root ) {
             return 0;
        }
        int leftHeight = 0, rightHeight = 0;
        
        TreeNode*leftSubtree  = root , *rightSubtree = root;
        
          while ( leftSubtree ) {
               leftHeight++;
                leftSubtree = leftSubtree ->left;
          }
         while ( rightSubtree ) {
                rightHeight++;
             rightSubtree = rightSubtree->right; 
         }
        
         if ( leftHeight == rightHeight ) {
            return pow(2,leftHeight) - 1;
         } 
             return  1 + countNodes( root->left ) + countNodes( root->right );
    }
};

File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int countNodes(TreeNode root) {
     int ans = 0;
     if(root == null) return 0;
      Queue<TreeNode>q = new LinkedList<>();
       q.add(root); 
        while(!q.isEmpty()) {
            int s = q.size();
            ans += s;
            for(int i = 0 ;i < s ;i++){
                  TreeNode cur = q.poll();
                   if(cur.left != null){
                      q.add(cur.left);
                   }
                    if(cur.right != null) {
                        q.add(cur.right);
                    }
            }
        }
        return ans;
    }
}